pkg/services/alerting/notifiers/slack.go

```diff
@@ -294,7 +294,10 @@ func (sn *SlackNotifier) Notify(evalContext *alerting.EvalContext) error {
 		"footer_icon": "https://grafana.com/assets/img/fav32.png",
 		"ts":          time.Now().Unix(),
 	}
-	if sn.NeedsImage() && imageURL != "" {
+	if sn.NeedsImage() {
+		if imageURL == "" {
+			imageURL = evalContext.ImageOnDiskPath
```

Comment:
This feels wrong to me and don't see how this would work. Have you been able to reproduce the problem described in the issue? There aren't really anything in the issue confirming that there is a bug or am I missing something?

Reply:
It feels wrong indeed. I was able to reproduce it and this fixes it though. That's why I wanted a second pair of 👀  since it seemed funny to me.

Reply:
Hmm okay. You're using it with or without a token? What's the value of `evalContext.ImageOnDiskPath` for you? At least before when using a token the image needed to be uploaded separately basically results in 2 posts in slack, one with the message and one with the image. You can see the code further down does this when token is configured.

ImagePublicURL should be populated correctly according to https://github.com/grafana/grafana/blob/07ef4060a34f557c8f0bee935e732c0963b57fbf/pkg/services/alerting/notifier.go#L231